src/meta/service/src/network.rs

```diff
@@ -125,21 +127,37 @@ impl RaftNetwork<LogEntry> for Network {
 
         let mut client = self.make_client(&target).await?;
 
-        let req = common_tracing::inject_span_to_tonic_request(rpc);
+        let mut last_err = None;
+        let mut mes = Default::default();
 
-        self.incr_meta_metrics_sent_bytes_to_peer(&target, req.get_ref());
+        for backoff in ExponentialBackoff::default() {
+            let req = common_tracing::inject_span_to_tonic_request(&rpc);
 
-        let resp = client.append_entries(req).await;
-        debug!("append_entries resp from: id={}: {:?}", target, resp);
+            self.incr_meta_metrics_sent_bytes_to_peer(&target, req.get_ref());
 
-        if resp.is_err() {
-            incr_meta_metrics_sent_failure_to_peer(&target);
+            let resp = client.append_entries(req).await;
+            debug!("append_entries resp from: id={}: {:?}", target, resp);
+
+            if let Err(e) = resp {
+                incr_meta_metrics_sent_failure_to_peer(&target);
+                last_err = Some(e);
+
+                // backoff and retry sending
+                sleep(backoff).await;
+                continue;
+            }
+            let resp = resp.unwrap();
+            mes = resp.into_inner();
+            // clean up
+            last_err = None;
```

Comment:
If you were gonna use both the Err and Ok, use a `match` would be better than a `unwrap()`